front/src/modules/trainschedule/components/ManageTrainSchedule/ManageTrainScheduleMap/ItineraryMarkers.tsx

```diff
@@ -65,49 +71,88 @@ const formatPointWithNoName = (
   </>
 );
 
-const extractMarkerInformation = (pathSteps: MarkerInformation[], showStdcmAssets: boolean) =>
-  pathSteps.reduce((acc, cur, index) => {
-    if (cur && cur.coordinates) {
-      if (index === 0) {
-        acc.push({
-          coordinates: cur.coordinates,
-          type: MARKER_TYPE.ORIGIN,
-          marker: cur,
-          imageSource: showStdcmAssets ? stdcmOrigin : originSVG,
-        });
-      } else if (index > 0 && index < pathSteps.length - 1) {
-        acc.push({
-          coordinates: cur.coordinates,
-          type: MARKER_TYPE.VIA,
-          marker: cur,
-          imageSource: showStdcmAssets ? stdcmVia : viaSVG,
-          index,
-        });
-      } else if (index === pathSteps.length - 1) {
-        acc.push({
-          coordinates: cur.coordinates,
-          type: MARKER_TYPE.DESTINATION,
-          marker: cur,
-          imageSource: showStdcmAssets ? stdcmDestination : destinationSVG,
-        });
+const extractMarkerInformation = (
+  pathSteps: MarkerInformation[],
+  showStdcmAssets: boolean,
+  suggestedOP: SuggestedOP[]
+): MarkerProperties[] =>
+  pathSteps
+    .map((pathStep, index): MarkerProperties | null => {
+      const matchingOp = suggestedOP.find((op) =>
+        matchPathStepAndOp(pathStep as PathItemLocation, op)
+      );
+
+      if (!matchingOp) return null;
+
+      if (pathStep && pathStep.coordinates) {
+        if (index === 0) {
+          return {
+            coordinates: pathStep.coordinates,
+            type: MARKER_TYPE.ORIGIN,
+            imageSource: showStdcmAssets ? stdcmOrigin : originSVG,
+            op: matchingOp,
+            pathStep,
+          };
+        }
+        if (index > 0 && index < pathSteps.length - 1) {
+          return {
+            coordinates: pathStep.coordinates,
+            type: MARKER_TYPE.VIA,
+            imageSource: showStdcmAssets ? stdcmVia : viaSVG,
+            index,
+            op: matchingOp,
+            pathStep,
+          };
+        }
```

Comment:
Not from your PR, but you could remove the `if` and move the return at the end since if it's not the first or last index, it can only be a via
```suggestion
          return {
            coordinates: pathStep.coordinates,
            type: MARKER_TYPE.VIA,
            imageSource: showStdcmAssets ? stdcmVia : viaSVG,
            index,
            op: matchingOp,
            pathStep,
          };
```

Reply:
Done